hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -146,7 +146,7 @@ private int addUpdateBucket(String partitionPath, String fileIdHint) {
    * @return smallFiles not in clustering
    */
   private List<SmallFile> filterSmallFilesInClustering(final Set<String> pendingClusteringFileGroupsId, final List<SmallFile> smallFiles) {
-    if (this.config.isClusteringEnabled()) {
```

Comment:
Good find. We can probably remove config.isClusteringEnabled() check below and only rely on pendingClusteringFileGroupsId?

cc: @lw309637554 any reason to base this on config originally in #2275?

Reply:
Hi @satishkotha Thanks for your view.  Only rely on pendingClusteringFileGroupsId is much better :) changed.

Reply:
@satishkotha @zhangyue19921010
At first we have two config for clustering. If set ASYNC_CLUSTERING_ENABLE_OPT_KEY will be ok.
  public boolean isAsyncClusteringEnabled() {
    return Boolean.parseBoolean(props.getProperty(HoodieClusteringConfig.ASYNC_CLUSTERING_ENABLE_OPT_KEY));
  }

  public boolean isClusteringEnabled() {
    // TODO: future support async clustering
    return inlineClusteringEnabled() || isAsyncClusteringEnabled();
  }

Reply:
@satishkotha @zhangyue19921010 
Use "if (!pendingClusteringFileGroupsId.isEmpty())" will improve ease of use. 
Another need to  modify. But if this will bring performance loss? @satishkotha 

"  private JavaRDD<HoodieRecord<T>> clusteringHandleUpdate(JavaRDD<HoodieRecord<T>> inputRecordsRDD) {
    if (config.isClusteringEnabled()) {"